server/src/test/java/io/crate/role/TransportRoleActionTest.java

```diff
@@ -160,6 +166,33 @@ public void test_drop_user_with_old_users_metadata() throws Exception {
         assertThat(res).isTrue();
     }
 
+    @Test
+    public void test_cannot_drop_user_mapped_to_foreign_servers() throws Exception {
+        Map<String, Role> roles = Map.of(
+            "role1", RolesHelper.roleOf("role1")
+        );
+        RolesMetadata rolesMetadata = new RolesMetadata(roles);
+        Metadata metadata = Metadata.builder()
+            .putCustom(RolesMetadata.TYPE, rolesMetadata)
+            .build();
+        ClusterState clusterState = ClusterState.builder(clusterService.state()).metadata(metadata).build();
```

Comment:
Other transport actions around role could follow the same refactoring as https://github.com/crate/crate/pull/15618.